db/index.js

```diff
@@ -971,3 +988,179 @@ function getIndexActions(indexes, existingItem, item, table) {
   })
   return {puts: puts, deletes: deletes}
 }
+
+// Relatively fast deep clone of simple objects/arrays
+function deepClone(obj) {
```

Comment:
Out of curiosity, why was it necessary to move this function and the one below?

Reply:
Good question! I originally wanted to reuse the `updateItem` action completely, but an update within a transaction has several fundamental difference from a single update:
- We have to check if there's a key conflict between an update in a transaction and any other query in the transaction. Single item updates by definition can't have conflicts.
- A transactional item can't be unlocked in the local database until all the operations are completed. Single item updates release their lock as soon as the one row is changed.
- Similarly, I'm using `batch` to perform the puts and deletes for transactions, I can't actually perform the update until every query in the transaction checks out as valid.

I did need these two functions, but I needed them separately from the rest of the `updateItem` action. Since they were being reused in multiple places and no longer "owned" by `updateItem`, I moved them up to `index`.